### PR TITLE
feature: build URLs from vercel data

### PR DESCRIPTION
**Pull request by jsdalton**
## Purpose

Our hosted app action should build real URLs by fetching appropriate data from the Vercel API and saved app installation parameters.

This PR implements the "construction" parts of the action. A later PR will fetch the data from the Vercel API.

## Approach

```
$ npm run call-app-action -- -a vercelGetPreviewEnvironments -s q5wv5ajna1af

> @contentful/vercel-app@1.0.0 call-app-action
> npx ts-node scripts/call-app-action.ts -a vercelGetPreviewEnvironments -s q5wv5ajna1af

APP ACTION RESULT
{
  sys: {
    id: '23yWjVk6iWcM3lYmemzCVj',
    type: 'AppActionCallDetails',
    space: { sys: { type: 'Link', linkType: 'Space', id: 'q5wv5ajna1af' } },
    sys: { type: 'Link', linkType: 'Environment', id: 'master' },
    createdBy: {
      sys: {
        type: 'Link',
        linkType: 'AppAction',
        id: 'vercelGetPreviewEnvironments'
      }
    },
    createdAt: '2024-04-01T15:25:10.635Z'
  },
  request: { function: 'actions/get-preview-envs.js', parameters: {} },
  response: {
    body: '{"ok":true,"data":[{"sys":{"id":"vercel-app-preview-environment","version":1,"type":"PreviewEnvironment"},"name":"Vercel App","description":"Content preview environment provided by Vercel app","configurations":[{"contentType":"blogPage","enabled":true,"example":false,"name":"","url":"https://team-integrations-vercel-playground-master.vercel.app/api/enable-draft?path=%2Fblogs%2F%7Bentry.fields.slug%7D&x-vercel-protection-bypass=ukkdTdqAgnG5DQHwFkIeQ22N1nUDWeU7","contentTypeFields":[]}],"envId":"vercel-app-preview-environment"}]}',
    statusCode: 200
  },
  statusCode: 200,
  responseAt: '2024-04-01T15:25:11.441Z',
  errors: [],
  eventType: 'call',
  url: 'hosted',
  requestAt: '2024-04-01T15:25:10.635Z',
  environmentId: 'master'
}
```

### CODE REVIEW

File: apps/vercel/app-actions/src/actions/get-preview-envs.spec.ts [hunk 1]
```diff
@@ -36,6 +21,7 @@ describe('get-preview-env.handler', () => {
 
   it('returns a list of preview environments', async () => {
     const result = await handler(callParameters, context);
+    console.log({ result });
```
Comment (matthew-gordon) [inline apps/vercel/app-actions/src/actions/get-preview-envs.spec.ts:24]: is this temporary or do we want to 🧹 ?

File: apps/vercel/app-actions/src/helpers/build-preview-urls-for-content-types.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  context: AppInstallationContextProps,
+  cmaClient: PlainClientAPI
+): Promise<PreviewUrlsForContentTypes> => {
+  const {
+    contentTypePreviewPathSelections: contentTypePreviewPaths,
+    vercelAccessToken,
+    selectedProject: projectId,
+  } = await fetchAppInstallationParameters(context, cmaClient);
+  const previewUrlParts = await constructVercelPreviewUrlParts(vercelAccessToken, projectId);
+  const apiPath = '/api/enable-draft'; // later we won't hard code this but get it from params
```
Comment (matthew-gordon) [inline apps/vercel/app-actions/src/helpers/build-preview-urls-for-content-types.ts:29]: 👍

**Review by matthew-gordon — APPROVED** [review]
LGTM, nothing blocking 👍 ❇️

File: apps/vercel/app-actions/src/actions/get-preview-envs.spec.ts (hunk 1 above)
Comment (jsdalton) [reply apps/vercel/app-actions/src/actions/get-preview-envs.spec.ts:24]: Yeah good catch. I will get rid of it in my fast follow PR.

**Pull request merged**